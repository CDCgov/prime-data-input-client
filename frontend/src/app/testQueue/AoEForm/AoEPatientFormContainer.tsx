import React from "react";
import { gql, useQuery } from "@apollo/client";
import { getFacilityIdFromUrl } from "../../utils/url";
import AoEForm from "./AoEForm";
import StepIndicator from "../../commonComponents/StepIndicator";

interface Props {
  patientId: string;
}
const GET_PATIENT = gql`
  query GetPatientDetails($id: String!) {
    patient(id: $id) {
      lookupId
      firstName
      middleName
      lastName
      birthDate
      street
      streetTwo
      city
      state
      zipCode
      telephone
      role
      email
      county
      race
      ethnicity
      gender
      residentCongregateSetting
      employedInHealthcare
      testResults {
        internalId
        dateTested
        result
      }
    }
  }
`;
const AoEPatientFormContainer = (props: Props) => {
  const { data, loading, error } = useQuery(GET_PATIENT, {
    variables: { id: props.patientId || "" },
    fetchPolicy: "no-cache",
  });

  if (loading) {
    return <p>Loading...</p>;
  }
  if (error) {
    return <p>error loading patient with id {props.patientId}...</p>;
  }

  const residentCongregateSetting = data.patient.residentCongregateSetting
    ? "YES"
    : "NO";

  const facilityId = getFacilityIdFromUrl();

  const employedInHealthcare = data.patient.employedInHealthcare ? "YES" : "NO";
  return (
    <main className="patient-app patient-app--form padding-bottom-4">
      <div className="grid-container maxw-tablet">
        <StepIndicator></StepIndicator>
        <AoEForm
          patient={{
            ...data.patient,
            residentCongregateSetting,
            employedInHealthcare,
          }}
          facilityId={facilityId}
          isModal={false}
          saveButtonText="Submit"
          onClose={null}
          saveCallback={null}
        />
      </div>
    </main>
  );
};

export default AoEPatientFormContainer;
