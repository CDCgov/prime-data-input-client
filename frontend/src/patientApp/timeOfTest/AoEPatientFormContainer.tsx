import React from "react";
import { useState } from "react";
import { getFacilityIdFromUrl } from "../../app/utils/url";
import AoEForm from "../../app/testQueue/AoEForm/AoEForm";
import StepIndicator from "../../app/commonComponents/StepIndicator";
import PatientProfile from "./PatientProfile";
import PatientForm from "../../app/patients/PatientForm";
import Button from "../../app/commonComponents/Button";
import { connect, useSelector } from "react-redux";
import { gql, useMutation } from "@apollo/client";

const PATIENT_LINK_SUBMIT_MUTATION = gql`
  mutation PatientLinkById(
    $plid: String!
    $birthDate: String!
    $pregnancy: String
    $symptoms: String
    $firstTest: Boolean
    $priorTestDate: String
    $priorTestType: String
    $priorTestResult: String
    $symptomOnset: String
    $noSymptoms: Boolean
  ) {
    patientLinkSubmit(
      internalId: $plid
      birthDate: $birthDate
      pregnancy: $pregnancy
      symptoms: $symptoms
      firstTest: $firstTest
      priorTestDate: $priorTestDate
      priorTestType: $priorTestType
      priorTestResult: $priorTestResult
      symptomOnset: $symptomOnset
      noSymptoms: $noSymptoms
    )
  }
`;

interface Props {
  patientId: string;
  page: string;
}

const AoEPatientFormContainer = ({ patientId, page }: Props) => {
  const patient = useSelector((state) => (state as any).patient as any);
  const facility = useSelector((state) => (state as any).facility as any);
  const plid = useSelector((state) => (state as any).plid as String);

  const residentCongregateSetting = patient.residentCongregateSetting
    ? "YES"
    : "NO";

  let facilityId = getFacilityIdFromUrl() || "";
  if (!facilityId && facility) {
    facilityId = facility.id;
  }
  const employedInHealthcare = patient.employedInHealthcare ? "YES" : "NO";

  const steps = [
    {
      label: "Profile information",
      value: "profile",
      order: 0,
      isCurrent: page === "profile",
    },
    {
      label: "Symptoms and history",
      value: "symptoms",
      order: 1,
      isCurrent: page === "symptoms",
    },
  ];

  const [isProfileEdit, setProfileEdit] = useState(false);
  const [submitMutation] = useMutation(PATIENT_LINK_SUBMIT_MUTATION);
  const saveCallback = (args: any) => {
    submitMutation({
      variables: {
        ...args,
        plid,
        birthDate: patient.birthDate,
      },
    });
  };

  const setProfileView = () => {
    window.scrollTo(0, 0);
    setProfileEdit(!isProfileEdit);
  }

  return (
    <main className="patient-app patient-app--form padding-bottom-4">
      <div className="grid-container maxw-tablet">
        <StepIndicator steps={steps} />
        {page === "symptoms" && (
          <AoEForm
            patient={{
              ...patient,
              residentCongregateSetting,
              employedInHealthcare,
            }}
            facilityId={facilityId}
            isModal={false}
            saveButtonText="Submit"
            noValidation={false}
            saveCallback={saveCallback}
          />
        )}
        {page === "profile" && (
          <>
            {!isProfileEdit && (
              <>
                <PatientProfile
                  patient={{
                    ...patient,
                    residentCongregateSetting,
                    employedInHealthcare,
                  }}
                />
                <Button
                  className="margin-top-1"
                  variant="outline"
                  label={"Edit information"}
                  onClick={setProfileView}
                />
              </>
            )}
            {isProfileEdit && (
              <div className={"class-name-here"}>
                <PatientForm
                  patient={{
                    ...patient,
                    residentCongregateSetting,
                    employedInHealthcare,
                  }}
                  activeFacilityId={facilityId}
                  patientId={patient.internalId}
                  isPxpView={true}
                  backCallback={setProfileView}
                />
              </div>
            )}
          </>
        )}
      </div>
    </main>
  );
};

export default connect()(AoEPatientFormContainer);
