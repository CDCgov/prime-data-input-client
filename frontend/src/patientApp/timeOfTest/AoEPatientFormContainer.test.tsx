import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import AoEPatientFormContainer from "./AoEPatientFormContainer";

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    listen: jest.fn(),
    push: jest.fn(),
  }),
}));

describe("AoEPatientFormContainer", () => {
  beforeAll(() => {
    jest
      .useFakeTimers("modern")
      .setSystemTime(new Date("2399-01-01").getTime());
  });
  it("snapshot", () => {
    const mockStore = configureStore([]);
    const store = mockStore({
      patient: {
        residentCongregateSetting: true,
        employedInHealthcare: true,
        birthDate: "",
      },
      plid: "foo",
    });
    const component = renderer.create(
      <Provider store={store}>
        <MockedProvider mocks={[]} addTypename={false}>
          <AoEPatientFormContainer page={""} />
        </MockedProvider>
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
