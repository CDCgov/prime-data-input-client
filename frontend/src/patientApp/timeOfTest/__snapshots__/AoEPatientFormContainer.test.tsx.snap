// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AoEPatientFormContainer snapshot 1`] = `
<main
  className="patient-app patient-app--form padding-bottom-4"
>
  <div
    className="grid-container maxw-tablet"
  >
    <div
      aria-label="progress"
      className="usa-step-indicator usa-step-indicator--counters-sm margin-y-205"
    >
      <ol
        className="usa-step-indicator__segments"
      >
        <li
          aria-current={false}
          className="usa-step-indicator__segment usa-step-indicator__segment--complete"
        >
          <span
            className="usa-step-indicator__segment-label"
          >
            Profile information
             
            <span
              className="usa-sr-only"
            >
              completed
            </span>
          </span>
        </li>
        <li
          aria-current={true}
          className="usa-step-indicator__segment usa-step-indicator__segment--current"
        >
          <span
            className="usa-step-indicator__segment-label"
          >
            Symptoms and history
             
            <span
              className="usa-sr-only"
            >
              completed
            </span>
          </span>
        </li>
      </ol>
      <div
        className="usa-step-indicator__header"
      >
        <h1
          className="usa-step-indicator__heading"
        >
          <span
            className="usa-step-indicator__heading-counter"
          >
            <span
              className="usa-sr-only"
            >
              Step
            </span>
            <span
              className="usa-step-indicator__current-step margin-right-05"
            >
              2
            </span>
            <span
              className="usa-step-indicator__total-steps"
            >
              of 
              2
            </span>
          </span>
          <span
            className="usa-step-indicator__heading-text"
          >
            Symptoms and history
          </span>
        </h1>
      </div>
    </div>
    <form
      onSubmit={[Function]}
    >
      <p>
        All fields marked with 
        <span
          className="usa-sr-only"
        >
          Required
        </span>
        <span
          className="usa-hint--required"
        >
           *
        </span>
         are required
      </p>
      <div
        className="prime-formgroup"
      >
        <fieldset
          className="usa-fieldset"
        >
          <legend
            className="prime-formgroup-heading usa-legend"
          >
            Symptoms
          </legend>
          <div
            className="usa-fieldset"
          >
            <fieldset
              className="usa-fieldset prime-checkboxes"
            >
              <legend
                className="usa-legend"
              >
                Are you experiencing any of the following symptoms?
                <span
                  className="usa-sr-only"
                >
                  Required
                </span>
                <span
                  className="usa-hint--required"
                >
                   *
                </span>
              </legend>
              <div
                className="usa-form-group"
              >
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-1"
                    name="no_symptoms"
                    onChange={[Function]}
                    type="checkbox"
                    value="no"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-1"
                  >
                    No Symptoms
                  </label>
                </div>
              </div>
            </fieldset>
            <div
              className="border-top-1px border-base-lighter margin-y-2"
            />
            <fieldset
              className="usa-fieldset prime-checkboxes symptom-checkboxes"
            >
              <legend
                className="usa-legend usa-sr-only"
              >
                What are your symptoms?
                <span
                  className="usa-hint"
                >
                   (optional)
                </span>
              </legend>
              <div
                className="usa-form-group"
              >
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-1"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="426000000"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-1"
                  >
                    Fever over 100.4F
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-2"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="103001002"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-2"
                  >
                    Feeling feverish
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-3"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="43724002"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-3"
                  >
                    Chills
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-4"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="49727002"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-4"
                  >
                    Cough
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-5"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="267036007"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-5"
                  >
                    Shortness of breath
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-6"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="230145002"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-6"
                  >
                    Difficulty breathing
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-7"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="84229001"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-7"
                  >
                    Fatigue
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-8"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="68962001"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-8"
                  >
                    Muscle or body aches
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-9"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="25064002"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-9"
                  >
                    Headache
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-10"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="36955009"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-10"
                  >
                    New loss of taste
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-11"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="44169009"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-11"
                  >
                    New loss of smell
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-12"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="162397003"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-12"
                  >
                    Sore throat
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-13"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="68235000"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-13"
                  >
                    Nasal congestion
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-14"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="64531003"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-14"
                  >
                    Runny nose
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-15"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="422587007"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-15"
                  >
                    Nausea
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-16"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="422400008"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-16"
                  >
                    Vomiting
                  </label>
                </div>
                <div
                  className="usa-checkbox"
                >
                  <input
                    checked={false}
                    className="usa-checkbox__input"
                    id="check-test-17"
                    name="symptom_list"
                    onChange={[Function]}
                    type="checkbox"
                    value="62315008"
                  />
                  <label
                    className="usa-checkbox__label"
                    htmlFor="check-test-17"
                  >
                    Diarrhea
                  </label>
                </div>
              </div>
            </fieldset>
          </div>
          <div
            className="usa-form-group"
          >
            <label
              className="usa-label"
              htmlFor="text-test-1"
            >
              Date of symptom onset
              <span
                className="usa-hint"
              >
                 (optional)
              </span>
            </label>
            <input
              aria-required="false"
              className="usa-input"
              id="text-test-1"
              max="2021-02-26"
              min="2020-02-01"
              name="symptom_onset"
              onChange={[Function]}
              type="date"
              value=""
            />
          </div>
        </fieldset>
      </div>
      <div
        className="prime-formgroup"
      >
        <fieldset
          className="usa-fieldset"
        >
          <legend
            className="prime-formgroup-heading usa-legend"
          >
            Test History
          </legend>
          <div
            className="prime-formgroup__wrapper"
          >
            <div
              className="usa-legend"
            >
              Is this your first COVID-19 test?
              <span
                className="usa-hint"
              >
                 (optional)
              </span>
            </div>
            <div
              className="prime-previous-test-display margin-top-2"
            >
              SimpleReport did not find any previous test data
            </div>
            <fieldset
              className="usa-fieldset prime-radios"
            >
              <legend
                className="usa-legend usa-sr-only"
              >
                Is this your first covid test?
                <span
                  className="usa-hint"
                >
                   (optional)
                </span>
              </legend>
              <div
                className="usa-form-group prime-radio--horizontal"
              >
                <div
                  className="usa-radio prime-radio--horizontal__container"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="radio-test-1_yes_0"
                    name="prior_test_flag"
                    onChange={[Function]}
                    type="radio"
                    value="yes"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="radio-test-1_yes_0"
                  >
                    Yes
                  </label>
                </div>
                <div
                  className="usa-radio prime-radio--horizontal__container"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="radio-test-1_no_1"
                    name="prior_test_flag"
                    onChange={[Function]}
                    type="radio"
                    value="no"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="radio-test-1_no_1"
                  >
                    No
                  </label>
                </div>
              </div>
            </fieldset>
          </div>
        </fieldset>
      </div>
      <div
        className="prime-formgroup"
      >
        <fieldset
          className="usa-fieldset"
        >
          <legend
            className="prime-formgroup-heading usa-legend"
          >
            Pregnancy
          </legend>
          <fieldset
            className="usa-fieldset prime-radios"
          >
            <legend
              className="usa-legend"
            >
              Currently pregnant?
              <span
                className="usa-hint"
              >
                 (optional)
              </span>
            </legend>
            <div
              className="usa-form-group"
            >
              <div
                className="usa-radio"
              >
                <input
                  checked={false}
                  className="usa-radio__input"
                  data-required="false"
                  disabled={false}
                  id="radio-test-1_77386006_0"
                  name="pregnancy"
                  onChange={[Function]}
                  type="radio"
                  value="77386006"
                />
                <label
                  className="usa-radio__label"
                  htmlFor="radio-test-1_77386006_0"
                >
                  Yes
                </label>
              </div>
              <div
                className="usa-radio"
              >
                <input
                  checked={false}
                  className="usa-radio__input"
                  data-required="false"
                  disabled={false}
                  id="radio-test-1_60001007_1"
                  name="pregnancy"
                  onChange={[Function]}
                  type="radio"
                  value="60001007"
                />
                <label
                  className="usa-radio__label"
                  htmlFor="radio-test-1_60001007_1"
                >
                  No
                </label>
              </div>
              <div
                className="usa-radio"
              >
                <input
                  checked={false}
                  className="usa-radio__input"
                  data-required="false"
                  disabled={false}
                  id="radio-test-1_261665006_2"
                  name="pregnancy"
                  onChange={[Function]}
                  type="radio"
                  value="261665006"
                />
                <label
                  className="usa-radio__label"
                  htmlFor="radio-test-1_261665006_2"
                >
                  Prefer not to answer
                </label>
              </div>
            </div>
          </fieldset>
        </fieldset>
      </div>
      <div
        className="margin-top-3"
      >
        <div
          className="sr-time-of-test-buttons display-flex flex-justify-end"
        >
          <button
            className="usa-button margin-right-0"
            id="aoe-form-save-button"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>
</main>
`;
